packages/material-ui/src/Accordion/Accordion.js

```diff
@@ -2,69 +2,102 @@ import * as React from 'react';
+    [`& .${accordionClasses.region}`]: styles.region,
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, square, expanded, disabled } = styleProps;
+
+  const slots = {
+    root: ['root', !square && 'rounded', expanded && 'expanded', disabled && 'disabled'],
+    region: ['region'],
+  };
+
+  return composeClasses(slots, getAccordionUtilityClass, classes);
+};
+
+const AccordionRoot = experimentalStyled(
+  Paper,
+  {},
+  {
+    name: 'MuiAccordion',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(({ theme, styleProps }) => {
   const transition = {
     duration: theme.transitions.duration.shortest,
   };
 
   return {
     /* Styles applied to the root element. */
-    root: {
-      position: 'relative',
-      transition: theme.transitions.create(['margin'], transition),
-      overflowAnchor: 'none', // Keep the same scrolling position
+    position: 'relative',
+    transition: theme.transitions.create(['margin'], transition),
+    overflowAnchor: 'none', // Keep the same scrolling position
+    '&:before': {
+      position: 'absolute',
+      left: 0,
+      top: -1,
+      right: 0,
+      height: 1,
+      content: '""',
+      opacity: 1,
+      backgroundColor: theme.palette.divider,
+      transition: theme.transitions.create(['opacity', 'background-color'], transition),
+    },
+    '&:first-of-type': {
       '&:before': {
-        position: 'absolute',
-        left: 0,
-        top: -1,
-        right: 0,
-        height: 1,
-        content: '""',
-        opacity: 1,
-        backgroundColor: theme.palette.divider,
-        transition: theme.transitions.create(['opacity', 'background-color'], transition),
+        display: 'none',
       },
-      '&:first-child': {
-        '&:before': {
-          display: 'none',
-        },
+    },
+    /* Styles applied to the root element if `expanded={true}`. */
+    [`&.${accordionClasses.expanded}`]: {
+      margin: '16px 0',
+      '&:before': {
+        opacity: 0,
       },
-      '&$expanded': {
-        margin: '16px 0',
-        '&:first-child': {
-          marginTop: 0,
-        },
-        '&:last-child': {
-          marginBottom: 0,
-        },
-        '&:before': {
-          opacity: 0,
-        },
+      '&:first-of-type': {
+        marginTop: 0,
+      },
+      '&:last-of-type': {
+        marginBottom: 0,
       },
-      '&$expanded + &': {
+      '& + &': {
         '&:before': {
           display: 'none',
         },
       },
-      '&$disabled': {
-        backgroundColor: theme.palette.action.disabledBackground,
-      },
+    },
+    /* Styles applied to the root element if `disabled={true}`. */
+    [`&.${accordionClasses.disabled}`]: {
+      backgroundColor: theme.palette.action.disabledBackground,
     },
     /* Styles applied to the root element unless `square={true}`. */
-    rounded: {
+    ...(!styleProps.square && {
       borderRadius: 0,
-      '&:first-child': {
+      '&:first-of-type': {
```

Comment:
This is also using the same selector. Now this one, **must** be moved to the first selector, as it would override it. Another potential fix is to use what we have in the Container component - https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Container/Container.js#L72 and apply the styles for the `square` that way (as a function callback)

Reply:
Sure thing, done!